File: Frontend/invcmsWeb/src/app/domain/consultant.ts
import { Company } from './company'
import { User } from './user'

export class Consultant {
    id: string
    identityNo:string
    user:User
    phone: string
    address: string
    ownedCompany: Company
    workedCompany: Company
    photo:string


    constructor() {
        this.id = null
        this.user=null
        this.phone = null
        this.address = null
        this.ownedCompany = null
        this.workedCompany = null
    }

}

File: Backend/src/main/java/com/ms/invcms/consultant/Consultant.java
package com.ms.invcms.consultant;


import com.ms.invcms.base.BaseEntity;
import com.ms.invcms.company.Company;
import com.ms.invcms.phone.Phone;
import com.ms.invcms.security.user_details.CustomUserDetails;
import com.ms.invcms.security.user_details.UserViewDto;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import org.codehaus.jackson.map.Serializers;

import javax.persistence.*;

@EqualsAndHashCode(callSuper = true)
@Entity
@Table(name="consultant")
@AllArgsConstructor
@NoArgsConstructor
@Data
public class Consultant extends BaseEntity {

    private String photo ;
    @OneToOne
    private Phone phone;
    private String address;
    private String identityNo;

    @OneToOne
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    private CustomUserDetails user;

    @ManyToOne
    private Company ownedCompany;

    @ManyToOne
    private Company workedCompany;


}
